clang/lib/AST/Type.cpp

```diff
@@ -1938,6 +1938,17 @@ TagDecl *Type::getAsTagDecl() const {
   return nullptr;
 }
 
+const TemplateSpecializationType *
+Type::getAsNonAliasTemplateSpecializationType() const {
+  for (const auto *T = this; /**/; /**/) {
+    const TemplateSpecializationType *TST =
+        T->getAs<TemplateSpecializationType>();
+    if (!TST || !TST->isTypeAlias())
+      return TST;
+    T = TST->desugar().getTypePtr();
```

Comment:
Can this ever result in a `nullptr`??  No, right?  So we don't have to worry about the next iteration of 1945 dereferencing nullptr?

Reply:
No, desugar never returns a null QualType.